cmd/promtool/tsdb.go

```diff
@@ -777,6 +779,36 @@ func formatSeriesSet(ss storage.SeriesSet) error {
 	return nil
 }
 
+func formatSeriesSetToJSON(ss storage.SeriesSet) error {
+	getSeriesID := func(in []byte) uint64 {
+		hash := fnv.New64()
+		_, _ = hash.Write(in)
+		return hash.Sum64()
+	}
+
+	seriesCache := make(map[uint64]struct{})
+	for ss.Next() {
+		series := ss.At()
+		lbs := series.Labels()
+
+		b, err := json.Marshal(lbs)
+		if err != nil {
+			return err
+		}
+
+		if len(b) == 0 {
+			continue
+		}
+
+		id := getSeriesID(b)
+		if _, ok := seriesCache[id]; !ok {
+			fmt.Printf("%s\n", string(b))
```

Comment:
This can be `PrintLn`, or just `os.Stdout.Write(b)` followed by a newline.

Reply:
fixed